### PR TITLE
feat: add devnets support

### PR DESCRIPTION
**Pull request by barnabasbusa**

### CODE REVIEW

**Comment by barnabasbusa** [comment]
Min config: 
```yaml
participants:
  - el_client_type: geth
    el_client_image: ethpandaops/geth:master-5b57727
    cl_client_type: lighthouse
    cl_client_image: ethpandaops/lighthouse:sidecar-inclusion-proof-c6be31c
    count: 1
network_params:
  network: "dencun-devnet-12"
additional_services: []
```

**Review by parithosh — APPROVED** [review]

File: src/shared_utils/shared_utils.star
Comment (h4ck3rk3y) [inline src/shared_utils/shared_utils.star:110]: but are they happy with 

"apple, mango, cat"

that should be do-able. I doublt they require a "list" type; a string should be enough?

File: src/prelaunch_data_generator/el_cl_genesis/el_cl_genesis_generator.star
Comment (h4ck3rk3y) [inline src/prelaunch_data_generator/el_cl_genesis/el_cl_genesis_generator.star:59]: can we make changes to the genesis generator so that the generation is in line with what we expect from dencun?

(i.e not have the mv etc here); not now for later

File: src/participant_network.star
Comment (h4ck3rk3y) [inline src/participant_network.star:169]: what is this magical "0"?

**Review by h4ck3rk3y — APPROVED** [review]

File: src/cl/lighthouse/lighthouse_launcher.star
Comment (barnabasbusa) [reply src/cl/lighthouse/lighthouse_launcher.star:365]: this PR is about devnets, the other one is about public testnets or mainnet

File: main.star
Comment (h4ck3rk3y) [inline main.star:72]: Can you change the README to show these new configurations?

File: src/cl/lodestar/lodestar_launcher.star
Comment (barnabasbusa) [reply src/cl/lodestar/lodestar_launcher.star:317]: there are weeks when we launch 4 different devnets. Trust me you don't want to keep track of all the devnet names.

File: src/prelaunch_data_generator/el_cl_genesis/el_cl_genesis_generator.star
Comment (barnabasbusa) [reply src/prelaunch_data_generator/el_cl_genesis/el_cl_genesis_generator.star:59]: We do the mv here too lol: https://github.com/ethpandaops/ansible-collection-general/blob/master/roles/ethereum_genesis/tasks/generate_genesis.yaml#L75

File: main.star
Comment (barnabasbusa) [reply main.star:72]: nobody should have a different experience to how to it was done before. It should be hidden from the user. We didn't specify how jwt was generated before. Why do you think we should do it now?

File: main.star
Comment (barnabasbusa) [reply main.star:175]: yes sir.

File: src/prelaunch_data_generator/el_cl_genesis/el_cl_genesis_generator.star
Comment (barnabasbusa) [reply src/prelaunch_data_generator/el_cl_genesis/el_cl_genesis_generator.star:59]: Changing the format upstream is a hassle, as lots of projects might potentially be using this. Id hold off with this unless its really needed.

File: src/participant_network.star
Comment (barnabasbusa) [reply src/participant_network.star:169]: its a dummy, genesis validators root. I can actually get this value and use it.

Comment (h4ck3rk3y) [reply src/participant_network.star:169]: can we use a constant? :P "DUMMY_GENESIS_VALIDATORS_ROOT = '0'"

File: main.star
Comment (h4ck3rk3y) [reply main.star:72]: I chose this random line but i meant the entire thing; the support of devnets and the ability to pass in `network_params.network`

Comment (barnabasbusa) [reply main.star:72]: ah yeah thats in the other PR already.

File: src/shared_utils/shared_utils.star
Comment (barnabasbusa) [reply src/shared_utils/shared_utils.star:110]: Lets do this after both this and #437 is merged in.

File: src/participant_network.star
Comment (barnabasbusa) [reply src/participant_network.star:169]: will solve in #437

**Pull request merged**
